### PR TITLE
Add argument for running training in `risk_mod` multiple times

### PR DESCRIPTION
**Pull request by janiavdv**
### Changes Implemented

- Add argument `n_train_runs` to `risk_mod` with default value 5 (ensures backwards compatibility)
- Add check for `n_train_runs` positive integer
- Refactor cyclical coordinate descent training to function`run_risk_mod` _within_ `risk_mod` (not exposed in the API)
  - `run_risk_mod` is called within a loop that has `n_train_runs` iterations
- Compare the accuracies of coefficients from training and return the model with the highest accuracy

### TODO

- [x] Correct accuracy calculation
- [x] Include initialization in each iteration (randomness)
- [x] Change AUC to objective function
- [x] Optimize loop by moving model score card to after `for` loop

### CODE REVIEW

File: R/risk_mod.R
Comment (janiavdv) [inline R/risk_mod.R:253]: **question**: Do we need to prevent users from setting the `seed` when `n_train_runs > 1`? Wouldn't setting the seed make the iterations to optimize accuracy redundant?

File: R/risk_mod.R
Comment (janiavdv) [inline R/risk_mod.R]: **question**: Is there a better way to check if an R numeric object is an integer? I was initially using `is.integer(n_train_runs)` but that method [seems faulty](https://stackoverflow.com/questions/3476782/check-if-the-number-is-integer)...

File: R/risk_mod.R
Comment (janiavdv) [inline R/risk_mod.R]: **note**: I realize this check is wrong and I'm working on the fix.

Comment (janiavdv) [reply R/risk_mod.R]: Resolved.

**Comment by janiavdv** [comment]
**note**: We need to squash and merge this PR once it's ready because the individual commits aren't functional on their own.

File: R/risk_mod.R
Comment (janiavdv) [inline R/risk_mod.R]: **todo**: change to train iteration with lowest objective function value

File: R/risk_mod.R
Comment (janiavdv) [inline R/risk_mod.R]: **todo**: add initialization to this function

File: R/risk_mod.R
Comment (janiavdv) [inline R/risk_mod.R]: **todo**: move this outside the loop so it only happens once at the end of `risk_mod()`

File: R/risk_mod.R
Comment (janiavdv) [inline R/risk_mod.R]: **question**: do we need `curr_mod` for `X`, `y`, `weights` and `lambda0`? Can't we just use the same ones from the original function call since they don't change in the `run_risk_mod()` initialization?

**Review by alicepaul — APPROVED** [review]

**Comment by janiavdv** [comment]
**note**: I had to `rebase` after this PR was merged. The singular commit that represents the code introduced by this PR is [here](https://github.com/hjeglinton/riskscores/commit/59d1a893db549eb6d885183ccd48d2e82d4ccbb7).

**Pull request merged**
